unstable/wlr-export-dmabuf-unstable-v1.xml

```diff
@@ -0,0 +1,201 @@
+        tv_sec which is a 64-bit value combined from tv_sec_hi and tv_sec_lo,
+        and the additional fractional part in tv_nsec as nanoseconds. Hence,
+        for valid timestamps tv_nsec must be in [0, 999999999].
+        The seconds part may have an arbitrary offset at start.
+      </description>
+      <arg name="tv_sec_hi" type="uint"
+           summary="high 32 bits of the seconds part of the timestamp"/>
+      <arg name="tv_sec_lo" type="uint"
+           summary="low 32 bits of the seconds part of the timestamp"/>
+      <arg name="tv_nsec" type="uint"
+           summary="nanoseconds part of the timestamp"/>
+    </event>
+
+    <enum name="cancel_reason">
+      <description summary="cancel reason">
+        Indicates reason for cancelling the frame.
+      </description>
+      <entry name="temporary" value="0"
+             summary="temporary error, source will produce more frames"/>
+      <entry name="permanent" value="1"
+             summary="fatal error, source will not produce frames"/>
+      <entry name="resizing" value="2"
+             summary="temporary error, source will produce frames"/>
+    </enum>
+
+    <event name="cancel">
+      <description summary="indicates the frame is no longer valid">
+        If the frame is no longer valid after the "frame" event has been called,
+        this callback will be used to inform the client to scrap the frame.
+        Source is still valid for as long as the subscription function does not
+        return NULL.
+        This may get called if for instance the surface is in the process of
+        resizing.
+      </description>
+      <arg name="reason" type="uint" enum="cancel_reason"
+           summary="indicates a reason for cancelling this frame capture"/>
+    </event>
+
+    <request name="destroy" type="destructor">
+      <description summary="delete this object, used or not">
+        Unreferences the frame, allowing it to be reused. Must be called as soon
+        as its no longer used.
+        Can be called at any time by the client after the "frame" event, after
+        which the compositor will not call any other events unless the client
+        resubscribes to capture more. The client will still have to close any
+        FDs it has been given.
+      </description>
+    </request>
+  </interface>
+
+  <interface name="zwlr_export_dmabuf_manager_v1" version="1">
+    <description summary="manager to inform clients and begin capturing">
+      This object is a manager with which to start capturing from sources.
+    </description>
+
+    <request name="capture_output">
+      <description summary="subscribe to start capturing">
+        Request to start capturing from an entire wl_output.
+      </description>
+      <arg name="frame" type="new_id"
+           interface="zwlr_export_dmabuf_frame_v1"/>
+      <arg name="overlay_cursor" type="int"
+           summary="include custom client hardware cursor on top of the frame"/>
```

Comment:
I think we should remove this

Reply:
Yeah, I'm still not sure how to implement it anyway. @ascent12, thoughts?

Reply:
Absolutely not, this is a required feature for many people who'd like to stream. Like I said, you can use the UNIVERSAL PLANES flag to get the final composited dmabuf which will have the cursor.

Reply:
Sure, we can get the buffer without the cursor. But getting the buffer without the cursor requires an extra compositing step if we have to fall back to software cursors even before dmabuf-export is considered.

Reply:
Actually, getting it with the cursor might also be nebulous, not all drivers support universal planes either.

Reply:
Fine if the description is changed to say that this might be inconsistent? It can't be made consistent (because clients can have their own soft cursor) but this option does need to stay as for the majority of cases it'll work fine.
Software cursors only happen in obscure edge cases anyway, and they're usually the cases where you want to have them overlaid.

Reply:
I have a laptop which doesn't support hardware cursors at all, it's not that obscure.

For v1 of this protocol I'd rather just skip this option and let the compositor do whatever it wants.

Reply:
I disagree, this reasoning is exactly why it took so long to have a stable and usable xdg_shell. And just because your laptop doesn't support it is no reason why the protocol has to be crippled. The bloody windows capture API has support for this as well, and it works in much the same way, if there's a software cursor it stays, otherwise you can choose between having a hardware cursor overlaid or not.

Reply:
Okay, so lets leave it up to the compositor then?

Reply:
Hmm, maybe not, I still think that if the compositor can it should respect the choice. If it can't, it'll just output whatever it can. Document this as such?

Reply:
I'm personally okay with that. It's not perfect, but the compositor will have a best effort strategy.

(We don't _need_ universal planes for this, we can just force software cursors if the client wants cursors, and (try to) use hardware cursors if not. It gets messy when multiple clients start capturing the same output though…)

Reply:
How so?

Reply:
If a client requests the output with the cursor and another one requests the output without it, we can't do both. That could happen if you're using a RDP client and screencasting at the same time for instance.